features/src/main/feature/camel-features.xml

```diff
@@ -560,10 +563,26 @@
         <bundle>mvn:org.apache.camel.karaf/camel-azure-servicebus/${project.version}</bundle>
     </feature>
     <feature name='camel-azure-storage-blob' version='${project.version}' start-level='50'>
+        <feature prerequisite="true">spifly</feature>
         <feature version='${camel-osgi-version-range}'>camel-core</feature>
         <feature version='[${azure-core-version},2)'>azure-storage</feature>
+        <feature version='[4.1,5)'>netty</feature>
+        <bundle dependency='true'>mvn:com.fasterxml.jackson.datatype/jackson-datatype-jsr310/${jackson2-version}</bundle>
+        <bundle dependency='true'>mvn:com.fasterxml.jackson.dataformat/jackson-dataformat-xml/${jackson2-version}</bundle>
         <bundle dependency='true'>wrap:mvn:com.azure/azure-storage-blob/${auto-detect-version}</bundle>
         <bundle dependency='true'>wrap:mvn:com.azure/azure-storage-blob-changefeed/${azure-storage-blob-changefeed-version}</bundle>
+        <bundle>wrap:mvn:com.azure/azure-core-http-netty/${auto-detect-version}</bundle>
```

Comment:
Have you tested the command `bundle:tree-show <id>` to see why the package is resolved when setting `dependency` to `true` in this case?

Reply:
karaf@root()> bundle:list | grep azure-core-http-netty
160 │ Active  │  50 │ 1.14.2         │ wrap_file__Users_ylopushen_.m2_repository_com_azure_azure-core-http-netty_1.14.2_azure-core-http-netty-1.14.2.jar_Bundle-Version_1.14.2_SPI-Provider__
karaf@root()> bundle:tree-show 160
Bundle wrap_file__Users_ylopushen_.m2_repository_com_azure_azure-core-http-netty_1.14.2_azure-core-http-netty-1.14.2.jar_Bundle-Version_1.14.2_SPI-Provider__ [160] is currently ACTIVE

wrap_file__Users_ylopushen_.m2_repository_com_azure_azure-core-http-netty_1.14.2_azure-core-http-netty-1.14.2.jar_Bundle-Version_1.14.2_SPI-Provider__ [160]
+- io.netty.codec-http [122]
|  +- io.netty.codec [120]
|  |  +- io.netty.common [125]
|  |  |  +- org.ops4j.pax.logging.pax-logging-api [4]
|  |  |     +- org.apache.karaf.services.eventadmin [2]
|  |  |        +- org.apache.felix.configadmin [12]
|  |  |        |  +- org.apache.felix.coordinator [8]
|  |  |        +- org.apache.felix.metatype [11]
|  |  +- io.netty.buffer [119]
|  |  |  +- io.netty.common [125]
|  |  +- io.netty.transport [130]
|  |     +- io.netty.common [125]
|  |     +- io.netty.buffer [119]
|  |     +- io.netty.resolver [128]
|  |        +- io.netty.common [125]
|  +- io.netty.transport [130]
|  +- io.netty.common [125]
|  +- io.netty.buffer [119]
|  +- io.netty.handler [126]
|     +- io.netty.codec [120]
|     +- io.netty.transport-native-unix-common [132]
|     |  +- io.netty.common [125]
|     |  +- io.netty.buffer [119]
|     |  +- io.netty.transport [130]
|     +- io.netty.transport [130]
|     +- io.netty.common [125]
|     +- io.netty.buffer [119]
|     +- bcprov [21]
|     +- bcpkix [20]
|     |  +- bcprov [21]
|     |  +- bcutil [22]
|     |     +- bcprov [21]
|     +- io.netty.resolver [128]
+- io.projectreactor.netty.reactor-netty-core [133]
|  +- io.netty.codec [120]
|  +- io.netty.codec-http [122]
|  +- io.netty.transport-native-unix-common [132]
|  +- io.netty.common [125]
|  +- io.netty.handler [126]
|  +- io.netty.transport-classes-epoll [131]
|  |  +- io.netty.common [125]
|  |  +- io.netty.buffer [119]
|  |  +- io.netty.transport-native-unix-common [132]
|  |  +- io.netty.transport [130]
|  +- io.projectreactor.reactor-core [135]
|  |  +- reactive-streams [148]
|  |  +- org.ops4j.pax.logging.pax-logging-api [4]
|  +- io.netty.handler-proxy [127]
|  |  +- io.netty.codec [120]
|  |  +- io.netty.codec-http [122]
|  |  +- io.netty.common [125]
|  |  +- io.netty.buffer [119]
|  |  +- io.netty.transport [130]
|  |  +- io.netty.codec-socks [124]
|  |     +- io.netty.codec [120]
|  |     +- io.netty.common [125]
|  |     +- io.netty.buffer [119]
|  |     +- io.netty.transport [130]
|  +- reactive-streams [148]
|  +- io.netty.transport [130]
|  +- io.netty.resolver-dns [129]
|  |  +- io.netty.codec [120]
|  |  +- io.netty.codec-dns [121]
|  |  |  +- io.netty.codec [120]
|  |  |  +- io.netty.common [125]
|  |  |  +- io.netty.buffer [119]
|  |  |  +- io.netty.transport [130]
|  |  +- io.netty.transport [130]
|  |  +- io.netty.common [125]
|  |  +- io.netty.buffer [119]
|  |  +- io.netty.handler [126]
|  |  +- io.netty.resolver [128]
|  +- io.netty.buffer [119]
|  +- io.netty.resolver [128]
+- io.netty.common [125]
+- io.netty.handler [126]
+- io.projectreactor.reactor-core [135]
+- io.netty.handler-proxy [127]
+- reactive-streams [148]
+- wrap_file__Users_ylopushen_.m2_repository_com_azure_azure-core_1.48.0_azure-core-1.48.0.jar_SPI-Consumer___Bundle-Version_1.48.0 [161]
|  +- com.fasterxml.jackson.datatype.jackson-datatype-jsr310 [114]
|  |  +- com.fasterxml.jackson.core.jackson-annotations [110]
|  |  +- com.fasterxml.jackson.core.jackson-core [111]
|  |  +- com.fasterxml.jackson.core.jackson-databind [112]
|  |     +- com.fasterxml.jackson.core.jackson-core [111]
|  |     +- com.fasterxml.jackson.core.jackson-annotations [110]
|  +- com.fasterxml.jackson.core.jackson-databind [112]
|  +- wrap_file__Users_ylopushen_.m2_repository_com_azure_azure-xml_1.0.0_azure-xml-1.0.0.jar_Bundle-Version_1.0.0 [157]
|  +- wrap_file__Users_ylopushen_.m2_repository_com_azure_azure-json_1.1.0_azure-json-1.1.0.jar_Bundle-Version_1.1.0 [153]
|  +- io.projectreactor.reactor-core [135]
|  +- reactive-streams [148]
|  +- org.ops4j.pax.logging.pax-logging-api [4]
|  +- com.fasterxml.jackson.core.jackson-annotations [110]
|  +- com.fasterxml.jackson.core.jackson-core [111]
|  +- com.fasterxml.jackson.dataformat.jackson-dataformat-xml [113]
|     +- stax2-api [149]
|     +- com.fasterxml.jackson.core.jackson-databind [112]
|     +- com.fasterxml.jackson.core.jackson-annotations [110]
|     +- com.fasterxml.jackson.core.jackson-core [111]
+- io.projectreactor.netty.reactor-netty-http [134]
|  +- io.netty.codec [120]
|  +- io.netty.codec-http [122]
|  +- io.netty.transport-native-unix-common [132]
|  +- io.projectreactor.netty.reactor-netty-core [133]
|  +- io.netty.common [125]
|  +- io.netty.handler [126]
|  +- io.projectreactor.reactor-core [135]
|  +- reactive-streams [148]
|  +- io.netty.codec-http2 [123]
|  |  +- io.netty.codec [120]
|  |  +- io.netty.codec-http [122]
|  |  +- io.netty.transport [130]
|  |  +- io.netty.common [125]
|  |  +- io.netty.buffer [119]
|  |  +- io.netty.handler [126]
|  +- io.netty.transport [130]
|  +- io.netty.buffer [119]
|  +- io.netty.resolver [128]
+- io.netty.transport [130]
+- io.netty.buffer [119]
+- io.netty.resolver [128]